src/java.desktop/unix/classes/sun/awt/screencast/ScreencastHelper.java

```diff
@@ -219,9 +219,8 @@ private static boolean checkReturnValue(int retVal) {
                     "Screen Capture in the selected area was not allowed"
             );
         } else if (retVal == ERROR) {
-            if (SCREENCAST_DEBUG) {
-                System.err.println("Screen capture failed.");
-            }
+            throw new RuntimeException(
+                "Screen capture has failed due to desktop environment error");
```

Comment:
Please remove this

Reply:
@prrace have you read my description of this PR? If so, and based on that, what can you propose to address the problem described?

Reply:
Such change requires a [CSR](https://wiki.openjdk.org/display/csr) .
If we want to get this fix into JDK 22, we don't have time for CSR, since JDK 22 will be forked this Thursday.

Reply:
It is beyond that. It is an incompatible change. It will be rejected forever.

Reply:
@prrace the related API already throw SecurityException if capture permission is not granted. moreover the API has been changed already in the pastr to accommodate that eg see

 https://docs.oracle.com/javase/8/docs/api/java/awt/Robot.html#getPixelColor-int-int-

VS

https://docs.oracle.com/en/java/javase/21/docs/api/java.desktop/java/awt/Robot.html#getPixelColor(int,int)

and so it stands to reason that a similar solution can be found for this problem as well if there is a will to do so.

An exception seem to be the most logical solution here to me however it could be something different, like the approach you describe in JDK-8321475 (i cant comment there so commenting here instead). Either way, like i said, i firmly believe that the current API behavior is just plain wrong and potentially harmful.